src/Microsoft.AspNetCore.Rewrite/RewriteOptions.cs

```diff
@@ -1,24 +1,49 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using Microsoft.Extensions.FileProviders;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Rewrite.Internal;
 
 namespace Microsoft.AspNetCore.Rewrite
 {
     /// <summary>
     /// Options for the <see cref="RewriteMiddleware"/> 
     /// </summary>
-    public class RewriteOptions
+    public class RewriteOptions : IRewriteOptionsInfrastructure
     {
+        public RewriteOptions()
+        {
+            ((IRewriteOptionsInfrastructure)this).DefaultConfigFileProvider = CreateConfigFileProvider();
```

Comment:
FWIW the explicit interface approach has precendent in [EF options building](https://github.com/aspnet/EntityFramework/blob/03bcb5122e3f577a84498545fcf130ba79a3d987/src/Microsoft.EntityFrameworkCore.InMemory/Extensions/InMemoryDbContextOptionsExtensions.cs#L67) APIs